bson/proxy.go

```diff
@@ -6,9 +6,9 @@
 
 package bson
 
-// Proxy is an interface implemented by types that cannot themselves be directly encoded. Types
+// proxy is an interface implemented by types that cannot themselves be directly encoded. Types
 // that implement this interface with have ProxyBSON called during the encoding process and that
 // value will be encoded in place for the implementer.
-type Proxy interface {
+type proxy interface {
 	ProxyBSON() (interface{}, error)
```

Comment:
We should consider completely removing the `ProxyBSON` feature. Based on [searching for uses in open-source repos](https://sourcegraph.com/search?q=context:global+lang:Go++-repo:%5Egithub%5C.com/mongodb/mongo-go-driver%24+-file:%28%5E%7C/%29vendor/+ProxyBSON&patternType=standard&case=yes&sm=0&groupBy=repo), it seems completely unused.